src/component/mod.rs

```diff
@@ -164,6 +164,18 @@ impl ComponentManager {
             active_components,
         }
     }
+
+    // The heartbeat function is called periodically
+    pub fn heartbeat(&self) {
+        for component in self.active_components.values() {
+            // It's okay not to block on the lock -- heartbeats have no guaranteed periodicity
+            // (Plus, this is only used for component shutdown, if someone has this lock, the
+            // component  is clearly still in use)
+            if let Some(mut handle) = component.try_lock() {
```

Comment:
Let's talk IRL about what this actually means, I think I can understand it from a "I'm reading code" standpoint but wouldn't be able to replicate it on my own